instrumentation/kafka-clients-metrics-3.0.0/src/main/java/com/nr/instrumentation/kafka/MetricsScheduler.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *
+ *  * Copyright 2023 New Relic Corporation. All rights reserved.
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+package com.nr.instrumentation.kafka;
+
+import com.newrelic.agent.bridge.AgentBridge;
+import com.newrelic.api.agent.NewRelic;
+import org.apache.kafka.common.metrics.KafkaMetric;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+
+import static com.nr.instrumentation.kafka.MetricsConstants.KAFKA_METRICS_DEBUG;
+import static com.nr.instrumentation.kafka.MetricsConstants.METRICS_AS_EVENTS;
+import static com.nr.instrumentation.kafka.MetricsConstants.METRICS_EVENT_TYPE;
+import static com.nr.instrumentation.kafka.MetricsConstants.METRIC_PREFIX;
+import static com.nr.instrumentation.kafka.MetricsConstants.REPORTING_INTERVAL_IN_SECONDS;
+
+public class MetricsScheduler {
+    private static final ScheduledThreadPoolExecutor executor = createScheduledExecutor();
+    private static final Map<NewRelicMetricsReporter, ScheduledFuture<?>> metricReporterTasks = new ConcurrentHashMap<>();
+
+    private MetricsScheduler() {}
+
+    public static void addMetricsReporter(NewRelicMetricsReporter metricsReporter) {
+        ScheduledFuture<?> task = executor.scheduleAtFixedRate(new MetricsSendRunnable(metricsReporter),
+                0L, REPORTING_INTERVAL_IN_SECONDS, TimeUnit.SECONDS);
+        metricReporterTasks.put(metricsReporter, task);
+    }
+
+    public static void removeMetricsReporter(NewRelicMetricsReporter metricsReporter) {
+        ScheduledFuture<?> task = metricReporterTasks.remove(metricsReporter);
+        task.cancel(false);
+    }
+
+    private static ScheduledThreadPoolExecutor createScheduledExecutor() {
+        return (ScheduledThreadPoolExecutor)Executors.newScheduledThreadPool(
+                1, new DefaultThreadFactory("NewRelicMetricsReporter-Kafka", true));
```

Comment:
To eliminate the `DefaultThreadFactory` class here and in the streams module, you could do what I did [here](https://github.com/newrelic/newrelic-java-agent/pull/1183/files) and supply an inline lambda for the thread factory.

```
Executors.newSingleThreadScheduledExecutor(runnable -> {
                final Thread thread = new Thread(runnable);
                thread.setDaemon(true);
                thread.setName(THREAD_NAME);
                return thread;
            }
```

Reply:
I updated my solution to use the inline lambda